docs/cookbook/getInstantWithLocalTimeInZone.mjs

```diff
@@ -0,0 +1,92 @@
+/**
+ * Get an absolute time corresponding with a calendar date / wall-clock time in
+ * a particular time zone, the same as Temporal.TimeZone.getAbsoluteFor() or
+ * Temporal.DateTime.inTimeZone(), but with more disambiguation options.
+ *
+ * As well as the default Temporal disambiguation options 'earlier', 'later',
+ * and 'reject', there are additional options possible:
+ *
+ * - 'earlierLater': Same as what the Moment Timezone and Luxon libraries do;
+ *   equivalent to 'earlier' when turning the clock back, and 'later' when
+ *   setting the clock forward.
+ * - 'clipEarlier': Equivalent to 'earlier' when turning the clock back, and
+ *   when setting the clock forward returns the time just before the clock
+ *   changes.
+ * - 'clipLater': Equivalent to 'later' when turning the clock back, and when
+ *   setting the clock forward returns the exact time of the clock change.
+ *
+ * @param {Temporal.DateTime} dateTime - Calendar date and wall-clock time to
+ *   convert
+ * @param {Temporal.TimeZone} timeZone - Time zone in which to consider the
+ *   wall-clock time
+ * @param {string} disambiguation - Disambiguation mode, see description.
+ * @returns {Temporal.Absolute} Absolute time in timeZone at the time of the
+ *   calendar date and wall-clock time from dateTime
+ */
+function getInstantWithLocalTimeInZone(dateTime, timeZone, disambiguation = 'earlier') {
+  // Handle the built-in modes first
+  if (['earlier', 'later', 'reject'].includes(disambiguation)) {
+    return timeZone.getAbsoluteFor(dateTime, { disambiguation });
+  }
+
+  const possible = timeZone.getPossibleAbsolutesFor(dateTime);
+
+  // Return only possibility if no disambiguation needed
+  if (possible.length === 1) return possible[0];
+
+  switch (disambiguation) {
+    case 'earlierLater':
+      if (possible.length === 0) return timeZone.getAbsoluteFor(dateTime, { disambiguation: 'later' });
+      return possible[0];
+    case 'clipEarlier':
+      if (possible.length === 0) {
+        const before = dateTime.minus({ hours: 1 }).inTimeZone(timeZone);
```

Comment:
The change above looks good, but you still have `{ hours: 1 }` hiding down here.  Is there a way to do this more generally?

Reply:
From testing this locally, it looks like the `.minus({ hours: 1 })` isn't even needed. The subsequent `.inTimeZone(timeZone)` call is using the default `earlier` disambiguation, so it always generates an absolute _before_ the transition.
I've tested it with the most extreme ends of transition ranges I could think of off the top of my head:

* `Australia/Lord_Howe` only changing clocks by 30 minutes for DST (using `2019-10-06T02:15`).
* `Pacific/Apia` skipping the entirety of December 30, 2011 by moving the International Date Line (using `2011-12-30T12:00`).

Both cases worked correctly with and without the `.minus({ hours: 1 })`.

Reply:
Ah, thanks, that's true — I've updated this (although spelling out the `disambiguation: 'earlier'` explicitly).

Reply:
> although spelling out the disambiguation: 'earlier' explicitly

Good move, it's now much easier to understand what it's doing at a glance.

Reply:
Much better!  Thanks!